virttest/qemu_devices/qbuses.py

```diff
@@ -650,6 +650,119 @@ def _increment_addr(self, addr, last_addr=None):
         return super(QPCIBus, self)._increment_addr(addr, last_addr=last_addr)
 
 
+class QPCIEBus(QPCIBus):
+    """
+    PCIE root bus representation (creates pcie-root-port for virtio pci
+    device by default.)
+    """
+
+    def __init__(self, busid, bus_type, root_type_list, aobject=None):
+        """
+        :param busid: id of the bus (pcie.0)
+        :param bus_type: type of the bus (PCIE)
+        :param aobject: Related autotest object (pci.0)
+        :param root_type_list: root port type list
+        """
+        super(QPCIEBus, self).__init__(busid, bus_type, aobject)
+        self.__root_ports = {}
+        self.__root_type_list = root_type_list
```

Comment:
I liked the `root_type` better as that is what you expect this bus to use, while `root_type_list` is just a list of all supported (in this case) root ports. Anyway even when the root port is not supported, it's still a root port, so IMO it doesn't make sense to treat unsupported root port differently, than a supported one.